admin/sql/CreateFunctions.sql

```diff
@@ -1255,6 +1255,86 @@ BEGIN
 END;
 $$ LANGUAGE 'plpgsql';
 
+-------------------------------------------------------------------
+-- Ratings
+-------------------------------------------------------------------
+
+CREATE OR REPLACE FUNCTION update_aggregate_rating(entity_type ratable_entity_type, entity_id INTEGER)
+RETURNS VOID AS $$
+BEGIN
+  -- update the aggregate rating for the given entity_id.
+  EXECUTE format(
+    $SQL$
+      UPDATE %2$I
+         SET rating = agg.rating,
+             rating_count = nullif(agg.rating_count, 0)
```

Comment:
The `nullif` preserves the behavior of the Perl, BTW, where no ratings = both the rating and rating_count columns are NULL.